app/controllers/Controller.scala

```diff
@@ -52,12 +55,65 @@ class Controller @Inject() (
     implicit request: Request[_] =>
       af.authorised() {
 
-        Future.successful(Ok("manageOrTrack"))
+        manageOrTrackView(vrn, manageOrTrackForm.fill(ManageOrTrack(None, vrn.value)))
+
+      }.recoverWith {
+        case e: AuthorisationException => authorisationException(e)
+      }
+  }
+
+  def manageOrTrackSubmit(): Action[AnyContent] = Action.async {
+    implicit request =>
+      af.authorised() {
+        val mtFormOptional = manageOrTrackForm.bindFromRequest()
+        mtFormOptional.value match {
+          case Some(mtForm) => mtForm.choice match {
+            case Some(choice) => {
+              choice match {
+                case ManageOrTrackOptions.vrt.value  => Redirect(routes.Controller.showResults(Vrn(mtForm.vrn)))
+                case ManageOrTrackOptions.bank.value => Future.successful(Ok(s"""TODO: manage_or_track_submit ${choice} ${mtForm.vrn} """))
+                case ManageOrTrackOptions.dd.value   => Future.successful(Ok(s"""TODO: manage_or_track_submit ${choice} ${mtForm.vrn} """))
+              }
+            }
+            case None => {
+              manageOrTrackView(Vrn(mtForm.vrn), manageOrTrackForm.fill(ManageOrTrack(None, mtForm.vrn)).withError("manage", ErrorMessages.`choose an option`.show))
+            }
+          }
+          case None => throw new RuntimeException("Could not get form from request")
+        }
+
       }.recoverWith {
         case e: AuthorisationException => authorisationException(e)
       }
   }
 
+  private def manageOrTrackView(vrn: Vrn, form: Form[ManageOrTrack])(
+      implicit
+      request: Request[_]): Future[Result] =
+    {
+
+      val customerDataF = desConnector.getCustomerData(vrn)
+      val nextUrl = for {
+        customerData <- customerDataF
+      } yield {
+        val bankDetailsExist = customerInformationFormatter.getBankDetailsExist(customerData)
+        val bankDetails: Option[BankDetails] = customerInformationFormatter.getBankDetails(customerData)
+
+        Ok(views.manage_or_track(vrn, false, bankDetailsExist, bankDetails, form))
+      }
+      nextUrl
```

Comment:
Let's inline it or rename the `nextUrl` to something more meaningful

Reply:
Done